core/src/codecs/default/DefaultIdBloomFilterFormat.cpp

```diff
@@ -36,8 +37,11 @@ DefaultIdBloomFilterFormat::read(const storage::FSHandlerPtr& fs_ptr, segment::I
 
     std::string dir_path = fs_ptr->operation_ptr_->GetDirectory();
     const std::string bloom_filter_file_path = dir_path + "/" + bloom_filter_filename_;
-    scaling_bloom_t* bloom_filter =
-        new_scaling_bloom_from_file(bloom_filter_capacity, bloom_filter_error_rate, bloom_filter_file_path.c_str());
+    scaling_bloom_t* bloom_filter{nullptr};
+    if (fs_ptr->operation_ptr_->CacheGet(bloom_filter_file_path)) {
+        bloom_filter =
+            new_scaling_bloom_from_file(bloom_filter_capacity, bloom_filter_error_rate, bloom_filter_file_path.c_str());
```

Comment:
The bloom_filter internally use mmap() to mapping a file handler. This could be a risk point.

Reply:
IMO, For reading bloom filter, it's OK, since the file if already exist, the segment writer should take responsible for it's correctness, but in DefaultIdBloomFilterFormat::create, CachePut may not valid, since the file content is not synced.

Reply:
You are right. 
In fact, in 0.11.0(experimental), we rewrite the bloom filter code, remove the mmap(). After the change, the read()/write() logic become more clear:
https://github.com/milvus-io/milvus/blob/experimental/core/src/codecs/IdBloomFilterFormat.cpp

To avoid this risk, we need to do same change in 1.x. But it is a next step work, not relate with this PR.